xdsl/dialects/csl/csl.py

```diff
@@ -1311,7 +1313,7 @@ class SetDsdStrideOp(IRDLOperation):
 
     name = "csl.set_dsd_stride"
     op = operand_def(DsdType)
-    stride = operand_def(IntegerType(8, Signedness.SIGNED))
+    stride = operand_def(eq(i8) | eq(IntegerType(8, Signedness.SIGNED)))
```

Comment:
```suggestion
    stride = operand_def(base(i8) | base(IntegerType(8, Signedness.SIGNED)))
```

Reply:
I'm a little surprised these don't raise errors

Reply:
true, and I'm now very confused about our constraints.

Reply:
Actually it's my mistake my suggested change is bad, I'll suggest another